packages/paste-core/components/menu/src/index.tsx

```diff
@@ -94,12 +107,51 @@ const Menu = React.forwardRef<HTMLDivElement, MenuProps>((props, ref) => {
 Menu.displayName = 'Menu';
 export {Menu};
 
-const MenuItem = React.forwardRef<HTMLDivElement, MenuItemProps>(({as = StyledMenuItem, ...props}, ref) => {
-  return <MenuPrimitiveItem {...props} as={as} ref={ref} />;
-});
+const MenuItem = React.forwardRef<HTMLDivElement, MenuItemProps>(
+  ({as = StyledMenuItem, variant: _variant, ...props}, ref) => {
+    const variant = _variant || React.useContext(MenuGroupContext);
+    return <MenuPrimitiveItem {...props} variant={variant} as={as} ref={ref} />;
+  }
+);
 MenuItem.displayName = 'MenuItem';
 export {MenuItem};
 
+const MenuGroup = React.forwardRef<HTMLDivElement, MenuGroupProps>(({label, icon, children, ...props}, ref) => {
+  return (
+    <MenuGroupContext.Provider value={MenuItemVariants.GROUP_ITEM}>
+      <Box {...safelySpreadBoxProps(props)} role="group" aria-label={label} textDecoration="none" ref={ref}>
+        <Box
+          display="flex"
+          alignItems="center"
+          paddingTop="space30"
+          paddingBottom="space30"
+          paddingLeft={icon != null ? 'space40' : 'space70'}
```

Comment:
paddingLeft when there's an icon in the MenuGroub label could be bumped up to space50